hack/verify-gen-crds.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+./hack/gen-crds.sh
+
+if ! git diff --exit-code config/crds.yml >/dev/null; then
```

Comment:
should we expand scope of the check? e.g. why are any of the files dirty in git at the end of the build process for example.

Reply:
There appears to be an issue when running `git diff -exit-code` in CI  on the whole project. It says `gnostic/OpenAPIv2` is being deleted. It may be related to our use of `replace github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1` in go.mod. We point gnostic 0.4.1 when [OpenAPIv2 was renamed to openapiv2](https://github.com/google/gnostic/pull/155). What is vendored in on develop is [OpenAPIv2](https://github.com/vmware-tanzu/carvel-kapp-controller/tree/7896992649b3247d284cca126cc488479dc120d5/vendor/github.com/googleapis/gnostic/OpenAPIv2). 

Not quite sure how to resolve the issue. For now, I have gone back to just checking that crds.yml is generated. Unless there is something simple I am missing, I would prefer to move forward with just checking crds.yml as this pr is mainly focused on fixing crd generation. I can open a separate issue for resolving the issue in CI mentioned above.

Reply:
Looks like this was a vendor issue. Hopefully this pr can catch these kinds of issues moving forward. I have gone ahead and added back the more comprehensive check.